test.rb

```diff
@@ -51,7 +52,7 @@
 
     end
 
-    describe "GET /products/#{$id}" do
+    describe "GET /products/{id}" do
```

Comment:
$

Reply:
- rimossa interpolazione superflua

quando viene interpolata la stringa $id non è definita (quindi nil) quindi non ha effetto:
```ruby
irb(main):001:0> "id è #{$id}";
=> "id è "
```
se vuole vederlo nel codice basta aggiungere`it { expect(3).to eq(2) }` dentro un blocco che è definito usando l'id e noterà che il nome del blocco non include l'id:
```
Failures:

  1) Test GET /products/ is expected to eq 2
     Failure/Error: it { expect(3).to eq(2) }

       expected: 2
            got: 3

       (compared using ==)
```

senza interpolazione secondo me è più chiaro:
```
Failures:

  1) Test GET /products/{id} is expected to eq 2
     Failure/Error: it { expect(3).to eq(2) }

       expected: 2
            got: 3

       (compared using ==)
```

Reply:
ok però allora usiamo :id al posto di {id}